databuilder/extractor/delta_lake_metadata_extractor.py

```diff
@@ -0,0 +1,316 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+from datetime import datetime
+import logging
+from collections import namedtuple
+
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.models.table_last_updated import TableLastUpdated
+from databuilder.models.table_metadata import TableMetadata, ColumnMetadata
+from pyhocon import ConfigFactory, ConfigTree  # noqa: F401
+from pyspark.sql import SparkSession
+from pyspark.sql.catalog import Table
+from typing import Iterator, Union, List, Dict, Optional  # noqa: F401
+
+TableKey = namedtuple('TableKey', ['schema', 'table_name'])
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ScrapedColumnMetadata(object):
```

Comment:
curious of why we don't reuse the ColumnMetadata model?

Reply:
So I initially tried to use ColumnMetadata, but ran into problems when also trying to get as much other metadata specific to delta-lake tables at the same time that doesn't fit well into the current ColumnMetadata class. An example (that actually is for hive tables in general) would be isPartitionColumn. If I add it to columnTags then it ends up tagging the table instead.

I think somebody is working on fixing column tagging, perhaps at that point I can use the tag dictionary to store all of this other metadata?